docs/cookbook.rst

```diff
@@ -0,0 +1,3 @@
+=================
+Rasterio Cookbook
```

Comment:
@perrygeo Is there an ideal format for cookbook entries? And, is it preferable if it relies only on `rasterio`, or would something using `fiona` and `shapely` as well be acceptable? The first application I had for all of these libraries (and by far my most common use case still) is masking rasters using shapefiles. I pieced together code @sgillies had posted in various places. I'd be happy to put it together into a recipe for that specific task.